modules/core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/BaseDocument.java

```diff
@@ -691,17 +691,13 @@ protected void setValueBlob(T state, Blob blob, String xpath) throws PropertyExc
     }
 
     protected void setPropertyBlobData(String xpath, String string) {
+        Blob blob = StringUtils.isNotEmpty(string) ? Blobs.createBlob(string) : null;
```

Comment:
Testing for "empty" is wrong, it's legal too have a blob containing a 0-byte content. 
Even though for now this is used in cases where the difference doesn't really matter (binary fulltext), it's not the case in general.